### PR TITLE
Use constants from external crate

### PR DESCRIPTION
**Pull request by n00m4d**
# What
This PR imports mpl-common-constants and uses a few constants from there.
# Why
To external crates should not import program crate itself but use either client crate or mpl-common-constants.

### CODE REVIEW

File: programs/bubblegum/program/src/processor/finalize_tree_with_root.rs [hunk 1]
```diff
@@ -126,11 +129,16 @@ pub(crate) fn check_stake<'info>(
         BubblegumError::MiningOwnerMismatch
     );
 
+    let realm_key =
+        Pubkey::from_str(DAO_PUBKEY).map_err(|_| BubblegumError::PubkeyReconstruction)?;
```
Comment (StanChe) [inline programs/bubblegum/program/src/processor/finalize_tree_with_root.rs]: we can't put the solana pubkey in that repo due to dependency hell? Or putting the binary version of it?

**Review by StanChe — COMMENTED** [review]
Fine with me, the only concerns are: 
- the constant keys is recreated from a string on any call, increasing the CU
- the key stays kind of invisible to the end user
- we'll most likely have some different keys in devnet and mainnet - maybe we'll have to add some feature flags

**Comment by n00m4d** [comment]
> Fine with me, the only concerns are:
> 
> * the constant keys is recreated from a string on any call, increasing the CU
> * the key stays kind of invisible to the end user
> * we'll most likely have some different keys in devnet and mainnet - maybe we'll have to add some feature flags

Almost all concerns are addressed, please check it

1. now not, data type of imported keys was changed
2. -
3. added feature to cargo.toml for that

**Review by StanChe — APPROVED** [review]

File: programs/bubblegum/program/Cargo.toml [hunk 2]
```diff
@@ -31,6 +33,7 @@ spl-associated-token-account = { version = ">= 1.1.3, < 3.0", features = ["no-en
 spl-token = { version = ">= 3.5.0, < 5.0", features = ["no-entrypoint"] }
 mplx-staking-states = { git = "https://github.com/adm-metaex/mplx-staking.git" }
 mplx-rewards = { git = "https://github.com/adm-metaex/mplx-rewards.git", features = ["no-entrypoint"] }
+mpl-common-constants = { git = "https://github.com/adm-metaex/mpl-common-constants.git", branch = "main" }
```
Comment (kstepanovdev) [inline programs/bubblegum/program/Cargo.toml:36]: Isn't `main` a default branch? Not an issue though, just curious

File: programs/bubblegum/program/tests/batch-mint.rs [hunk 3]
```diff
@@ -182,8 +182,8 @@ async fn initialize_staking_accounts(
 
     let registrar = Registrar {
         governance_program_id: governance_program_id.clone(),
-        realm: REALM,
-        realm_governing_token_mint: REALM_GOVERNING_MINT,
+        realm: Pubkey::new_from_array(DAO_PUBKEY),
```
Comment (kstepanovdev) [inline programs/bubblegum/program/tests/batch-mint.rs:185]: Doesn't just `into()` work?

File: programs/bubblegum/program/src/error.rs [hunk 4]
```diff
@@ -108,6 +108,8 @@ pub enum BubblegumError {
     StakingVoterAuthorityMismatch,
     #[msg("Invalid mining owner")]
     MiningOwnerMismatch,
+    #[msg("Cannot create pubkey from string")]
```
Comment (kstepanovdev) [inline programs/bubblegum/program/src/error.rs:111]: Do we still need this error? Seems like a relict from times when Pubkeys were strings.

**Review by kstepanovdev — APPROVED** [review]
lgtm

**Pull request merged**